bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/tools/DelayedBatchProcessing.java

```diff
@@ -56,18 +57,19 @@ public DelayedBatchProcessing(int delay, Consumer<List<T>> consumer, ScheduledEx
     }
 
     /**
-     * Add new object to the batch process list. If the list was empty, the delay timer
-     * is armed and all successive objects are accumulated from here on.
+     * Add new object to the batch process list. Every time a new object is received,
+     * the delay timer is rescheduled.
      *
      * @param t An object
      */
     @Override
-    public void accept(T t) {
+    synchronized public void accept(T t) {
```

Comment:
The AtomicReference handles your synchronization so you can remove the `synchronized` here
```suggestion
    public void accept(T t) {
```

Reply:
In my defense, I can say that in the first push it was removed... :facepalm: